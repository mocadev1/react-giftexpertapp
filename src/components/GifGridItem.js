import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ( { title, url} ) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

export default GifGridItem

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


/**
 * Implementando Proptypes y actualizar snapshots
 * 
 * 1. AÃ±adir PropTypes url, title -> Obligatorios
 * 2. Enviar url y title a la hora de utilizar shallow()
 * 3. Actualizar snapshot con los cambios 
 */