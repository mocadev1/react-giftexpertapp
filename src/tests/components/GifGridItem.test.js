import { shallow } from "enzyme"


import GifGridItem from "../../components/GifGridItem"


describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'Un titulo de test';
    const url = 'https://localhost/imagen.jpg';
    
    const img = {
        title,
        url
    }

    const wrapper = shallow(<GifGridItem { ...img } />);
    
    // beforeEach( () => {
    //     wrapper = shallow( <GifGridItem { ...img } />);
    // });
    

    test('debe de renderizar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
        
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        
        expect( img.prop('alt') ).toBe(title);
        expect( img.prop('src') ).toBe(url);
        
    });

    test('debe de tener la clase animate__fadeIn', () => {

        const div = wrapper.find('div');

        expect( div.prop('className') ).toContain('animate__fadeIn');
    })
    
    
    
    
})