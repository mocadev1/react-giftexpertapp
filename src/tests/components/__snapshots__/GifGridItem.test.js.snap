// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo de test"
    src="https://localhost/imagen.jpg"
  />
  <p>
     
    Un titulo de test
     
  </p>
</div>
`;
